### ISSUE TITLE
Protocol for Hrishi | Shocking behaviour

### ISSUE THREAD

**Issue by dilawar** [BhallaLab/MouseBehaviour#47]
- [x] Integrate 4 channels solid state relays for switching b/w input to Arduino/Shocker and readout from arduino. These are mutually exclusive.

**Comment by dilawar** [comment]
Connections on June 27, 2019. 

![IMG_20190628_100637](https://user-images.githubusercontent.com/895681/60318796-f9194000-9991-11e9-970a-5b3b2455e5d4.jpg)

**Comment by dilawar** [comment]
[2_solid_relay.pdf](https://github.com/BhallaLab/MouseBehaviour/files/3341734/2_solid_relay.pdf)

Datasheets of solid state relays used.

![](https://cdn.shopify.com/s/files/1/1978/9859/products/1_6d7cf2f2-e8ae-4812-9d60-4652ad137560_1024x1024.jpg?v=1508147488)

**Comment by dilawar** [comment]
@hrishi16 I made a partial schematic. I'd need your help to finish it. The project is here: https://easyeda.com/dilawar.s.rajput/shocking-protocol 

![Schematic_Shocking-Protocol_Schematic_20190629172237](https://user-images.githubusercontent.com/895681/60383744-9bb8e800-9a92-11e9-9cba-e67261ca0660.png)

**Comment by dilawar** [comment]
@hrishi16 Prepare of beta testing with everything connected. Once beta is done, I'll merge it to master. 

Here are the main changes: (@ananthamurthy Keeping you in loop).

- Frame displayed is color but the frame which is saved is b/w. 
- Shock data is plotted on red channel. Eyeblink data is plotted on blue channel. 
![Screenshot from 2019-07-03 15-48-48](https://user-images.githubusercontent.com/895681/60584093-e8aa0080-9da9-11e9-85e2-38bcffab4a01.png)
- One extra field is added before rotary encoder values. 3rds from the last is the shock value.

**Comment by dilawar** [comment]
@ananthamurthy Let me know if you have read about these changes.

**Comment by dilawar** [comment]
Merged into master by #48 .
